napari/layers/points/points.py

```diff
@@ -336,13 +359,41 @@ def __init__(
 
         self.refresh_colors()
 
-        # set the current_* properties
-        self._current_edge_color = self.edge_color[-1]
-        self._current_face_color = self.face_color[-1]
         self.size = size
-        self.current_properties = {
-            k: np.asarray([v[-1]]) for k, v in self.properties.items()
-        }
+        # set the current_* properties
+        if len(data) > 0:
+            self._current_edge_color = self.edge_color[-1]
+            self._current_face_color = self.face_color[-1]
+            self.current_properties = {
+                k: np.asarray([v[-1]]) for k, v in self.properties.items()
+            }
+        elif len(data) == 0 and self.properties:
+            self.current_properties = {
+                k: np.asarray([v[0]])
+                for k, v in self.default_properties.items()
+            }
+            if self._edge_color_mode == ColorMode.DIRECT:
+                self._current_edge_color = transform_color_with_defaults(
+                    num_entries=1,
+                    colors=edge_color,
+                    elem_name="edge_color",
+                    default="white",
+                )
+            else:
+                self._current_edge_color = np.array([0, 0, 0, 1])
+            if self._face_color_mode == ColorMode.DIRECT:
+                self._current_face_color = transform_color_with_defaults(
+                    num_entries=1,
+                    colors=face_color,
+                    elem_name="face_color",
+                    default="white",
+                )
+            else:
+                self._current_face_color = np.array([1, 1, 1, 1])
+        else:
+            self._current_edge_color = self.edge_color[-1]
+            self._current_face_color = self.face_color[-1]
+            self.current_properties = {}
```

Comment:
I'm quite wary of big if/else cascades with layers of nesting. At the very least, this should be abstracted away into its own function. But it would be worth spending some time to try to simplify it. For example, we follow the exact same code path for both edge and face color, with a few things changed, so it might be worth writing `self._set_current_color(element='face', color=face_color)`, then use `getattr` to grab the `f'_{element}_color_mode'` element as required, and set the `f'_current_{element}_color'` attribute.